File: virtual/AtCoder/viral-03/01.py
# @uni_kakurenbo
# https://github.com/uni-kakurenbo/competitive-programming-workspace
#
# CC0 1.0  http://creativecommons.org/publicdomain/zero/1.0/deed.ja

# #language PyPy3 #

from sys import setrecursionlimit, stderr, argv
# setrecursionlimit(10**5)
def debug(*args, **opts):
    if argv[-1] == "LOCAL_JUDGE": print(*args, **opts, file=stderr)

n = int(input())

ans = 10**100
for i in range(1, n):
    a = sum(int(d) for d in str(i))
    b = sum(int(d) for d in str(n-i))
    debug(i, n-i, a, b)
    ans = min(ans, a+b)
print(ans)

File: practice/AtCoder/practice-000/01.cpp
/*
 * @uni_kakurenbo
 * https://github.com/uni-kakurenbo/competitive-programming-workspace
 *
 * CC0 1.0  http://creativecommons.org/publicdomain/zero/1.0/deed.ja
 */
/* #language C++ GCC */
/* #region template */
#include <bits/stdc++.h>
using namespace std;

#include "template.hpp"
/* #endregion */

signed main() {
    int n; cin >> n;
    vector<int> a(n); input >> a;

    map<int,int> cnt;
    ll sum = 0;
    ITR(v, a) { ++cnt[v], sum += v; }

    int q; cin >> q;
    REP(q) {
        int b, c; cin >> b >> c;
        sum -= 1LL * cnt[b] * b;
        sum += 1LL * cnt[b] * c;
        cnt[c] += cnt[b];
        cnt[b] = 0;
        print(sum);
    }



    return 0;
}
